swimster-ui/src/contexts/auth.js

```diff
@@ -1,13 +1,51 @@
 import { useState, createContext, useContext } from 'react'
+import apiClient from '../services/apiClient'
 
 const AuthContext = createContext(null)
 
 export const AuthContextProvider = ({ children }) => {
+    const [initialized, setInitialized] = useState(false)
     const [user, setUser] = useState({})
+    const [error, setError] = useState(null)
+
+    // auth functions from apiClient w/ validation and security checks
+    const handlers = {
+        loginUser: async credentials => {
+            const { data, error } = await apiClient.loginUser(credentials)
+            if (data?.user) {
+                setUser(data.user)
+                apiClient.setToken(data.token)
+                localStorage.setItem("swimster_session_token", data.token)
+            } if (error) setError(error)
+        },
+        registerUser: async credentials => {
+            const { data, error } = await apiClient.registerUser(credentials)
+            if (data?.user) {
+                setUser(data.user)
+                apiClient.setToken(data.token)
+                localStorage.setItem("swimster_session_token", data.token)
+            } if (error) setError(error)
+        },
+        logoutUser: () => {
+            apiClient.logoutUser()
+            setUser({})
+            setInitialized(true)
```

Comment:
why do you setInitialized here when a user logs out?

Reply:
Ooh, good find. I meant to set it to `false`. The `initialized` state variable is to log when all of the context from `auth`, `listings`, and `reservations` is mounted and unmounted when logging in/out

Reply:
done!